storage/src/node/branch.rs

```diff
@@ -6,6 +6,18 @@ use serde::{Deserialize, Serialize};
 use crate::{LeafNode, LinearAddress, Path, TrieHash};
 use std::fmt::{Debug, Error as FmtError, Formatter};
 
+#[derive(PartialEq, Eq, Clone, Serialize, Deserialize, Default, Debug, Copy)]
+/// A child of a branch node.
+pub enum Child {
+    #[default]
+    /// There is no child at this index.
+    None,
+    /// We know the child's address but not its hash.
+    Address(LinearAddress),
+    /// We know the child's address and hash.
+    AddressWithHash(LinearAddress, TrieHash),
+}
+
```

Comment:
If you do this, you can eliminate a lot of match statements elsewhere:
```suggestion
impl Child {
  fn address(&self) -> Option<LinearAddress> {
    match self {
        None => None,
        Address(addr) => Some(addr),
        AddressWithHash(addr, _) => Some(addr),
    }
  }
```